### ISSUE TITLE
Clarify how to create and update the release branch in the release procedure

### ISSUE THREAD

**Issue by kjun9** [stellargraph/stellargraph#953]
### Description

Currently the release procedure explains how to create a release branch locally, and what changes need to be made, but the exact procedure for making updates could be clarified by including some more details such as:

Add a line to push the branch
```
git checkout -b release-X.X.X develop
git push -u origin release-X.X.X
```

Add detail about how to make a change
> To make updates on the release branch, create a new feature branch with the required changes and make a PR to merge into the release branch so that they can be approved by other team members.

Add detail about a PR into master for approval purposes
> Once the release branch is ready to be merged into master, create a PR into master, but **do not use this to actually merge**, only use for reviewing purposes.
